src/main/scala/com/monsanto/arch/cloudformation/model/resource/ApplicationAutoScaling.scala

```diff
@@ -71,6 +71,64 @@ object ApplicationAutoScaling {
     private type T = StepScalingPolicyConfiguration
     implicit val format: JsonFormat[T] = jsonFormat5(apply)
   }
+
+  case class CustomizedMetricSpecification(Dimensions: Option[Seq[`AWS::CloudWatch::Alarm::Dimension`]] = None,
+                                           MetricName: String,
+                                           Namespace: `AWS::CloudWatch::Alarm::Namespace`,
+                                           Statistic: `AWS::CloudWatch::Alarm::Statistic`,
+                                           Unit: Option[`AWS::CloudWatch::Alarm::Unit`] = None)
+
+  object CustomizedMetricSpecification extends DefaultJsonProtocol {
+    implicit val format: JsonFormat[CustomizedMetricSpecification] = jsonFormat5(apply)
+  }
+
+  sealed trait PredefinedMetricType
+  object PredefinedMetricType extends DefaultJsonProtocol {
+    object DynamoDBReadCapacityUtilization extends PredefinedMetricType
+    object DynamoDBWriteCapacityUtilization extends PredefinedMetricType
+    object ALBRequestCountPerTarget extends PredefinedMetricType
+    object RDSReaderAverageCPUUtilization extends PredefinedMetricType
+    object RDSReaderAverageDatabaseConnections extends PredefinedMetricType
+    object EC2SpotFleetRequestAverageCPUUtilization extends PredefinedMetricType
+    object EC2SpotFleetRequestAverageNetworkIn extends PredefinedMetricType
+    object EC2SpotFleetRequestAverageNetworkOut extends PredefinedMetricType
+    object SageMakerVariantInvocationsPerInstance extends PredefinedMetricType
+    object ECSServiceAverageCPUUtilization extends PredefinedMetricType
+    object ECSServiceAverageMemoryUtilization extends PredefinedMetricType
+
+    val values = Seq(
+      DynamoDBReadCapacityUtilization,
+      DynamoDBWriteCapacityUtilization,
+      ALBRequestCountPerTarget,
+      RDSReaderAverageCPUUtilization,
+      RDSReaderAverageDatabaseConnections,
+      EC2SpotFleetRequestAverageCPUUtilization,
+      EC2SpotFleetRequestAverageNetworkIn,
+      EC2SpotFleetRequestAverageNetworkOut,
+      SageMakerVariantInvocationsPerInstance,
+      ECSServiceAverageCPUUtilization,
+      ECSServiceAverageMemoryUtilization)
+
+    implicit val format: JsonFormat[PredefinedMetricType] = new EnumFormat[PredefinedMetricType](values)
+  }
+
+  case class PredefinedMetricSpecification(PredefinedMetricType: PredefinedMetricType,
+                                           ResourceLabel: Option[Token[String]] = None)
+
+  case object PredefinedMetricSpecification extends DefaultJsonProtocol {
+    implicit val format: JsonFormat[PredefinedMetricSpecification] = jsonFormat2(apply)
+  }
+
+  case class TargetTrackingScalingPolicyConfiguration(CustomizedMetricSpecification: Option[CustomizedMetricSpecification] = None,
+                                                      DisableScaleIn: Option[Boolean] = None,
+                                                      PredefinedMetricSpecification: Option[String] = None,
```

Comment:
Shouldn't this be `Option[PredefinedMetricSpecification]`?

Reply:
Good catch. Thanks for fixing this.